src/Test/Perf/Runner/Benchview.cs

```diff
@@ -1,152 +1,164 @@
 using System.IO;
-using System.Xml;
+
+using Roslyn.Test.Performance.Utilities;
 using static Roslyn.Test.Performance.Utilities.TestUtilities;
 using static Roslyn.Test.Performance.Runner.Tools;
 
 namespace Roslyn.Test.Performance.Runner
 {
     public static class Benchview 
     {
-        private const string s_BenchviewPath = @"\\vcbench-srv4\benchview\uploads\vibench";
+        const string sasEnvVar = "BV_UPLOAD_SAS_TOKEN";
+        static string scriptDir = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Microsoft.BenchView.JSONFormat", "tools");
+        static string outputDir = GetCPCDirectoryPath();
+        static string[] validSubmissionTypes = new string[] { "rolling", "private", "local" };
 
-        internal static void UploadBenchviewReport()
+        public static string[] ValidSubmissionTypes
         {
-            // Convert the produced consumptionTempResults.xml file to consumptionTempResults.csv file
-            var elapsedTimeCsvFilePath = Path.Combine(GetCPCDirectoryPath(), "consumptionTempResults_ElapsedTime.csv");
-            var result = ConvertConsumptionToCsv(Path.Combine(GetCPCDirectoryPath(), "consumptionTempResults.xml"), elapsedTimeCsvFilePath, "Duration_TotalElapsedTime");
-
-            if (result)
+            get
             {
-                var elapsedTimeViBenchJsonFilePath = GetViBenchJsonFromCsv(elapsedTimeCsvFilePath, null, null);
-                string jsonFileName = Path.GetFileName(elapsedTimeViBenchJsonFilePath);
-
-                Log("Copying the json file to the share");
-                File.Copy(elapsedTimeViBenchJsonFilePath, Path.Combine(s_BenchviewPath, jsonFileName));
-                Log("Done Copying");
+                return validSubmissionTypes;
             }
-            else
+        }
+
+        internal static bool IsValidSubmissionType(string submissionType)
+        {
+            foreach(var type in validSubmissionTypes)
             {
-                Log("Conversion from Consumption to csv failed.");
+                if(submissionType == type)
+                {
+                    return true;
+                }
             }
+
+            return false;
         }
 
-        /// Takes a consumptionTempResults file and converts to csv file
-        /// Each info contains the {ScenarioName, Metric Key, Metric value}
-        private static bool ConvertConsumptionToCsv(string source, string destination, string requiredMetricKey)
+        internal static void CheckEnvironment()
         {
-            Log("Entering ConvertConsumptionToCsv");
-            if (!File.Exists(source))
+            var sasToken = Environment.GetEnvironmentVariable(sasEnvVar);
+            if (String.IsNullOrEmpty(sasToken))
             {
-                Log($"File {source} does not exist");
-                return false;
+                throw new Exception($"{sasEnvVar} was not defined");
             }
 
-            try
+            var whereGit = ShellOut("where", "git");
```

Comment:
For instances where `ShellOutVital` *needs* to succeed for the program to continue, consider `ShellOutVital`.